packages/esm-patient-medications-app/src/api/api.ts

```diff
@@ -70,11 +103,45 @@ export function getOrderTemplatesByDrug(drugUuid: string, abortController?: Abor
   });
 }
 
-export function getCurrentOrderBasketEncounter(patientUuid: string, abortController?: AbortController) {
+export function useCurrentOrderBasketEncounter(patientUuid: string) {
   const [nowDateString] = new Date().toISOString().split('T');
-  return openmrsFetch(`/ws/rest/v1/encounter?patient=${patientUuid}&fromdate=${nowDateString}&limit=1`, {
-    signal: abortController?.signal,
-  });
+  const sessionObject = useSession();
+  const config = useConfig() as ConfigObject;
+  const { data, error, mutate } = useSWR<FetchResponse<{ results: Array<any> }>, Error>(
+    `/ws/rest/v1/encounter?patient=${patientUuid}&fromdate=${nowDateString}&limit=1`,
+    openmrsFetch,
+  );
+
+  useEffect(() => {
```

Comment:
Actually, the error comes up in the notification, since there's no error message handling in the UI.

The reason for wrapping in the useEffect is that on re-rendering. and if the error object is still there, it'll show the notification again and again.